app/controllers/course/assessment/submission/submissions_controller.rb

```diff
@@ -298,6 +302,24 @@ def log_service
       Course::Assessment::SessionLogService.new(@assessment, session, @submission)
   end
 
+  def monitoring_component_enabled
+    current_component_host[:course_monitoring_component]
+  end
+
+  def monitoring_service
+    return nil unless should_monitor?
+
+    @monitoring_service ||= Course::Assessment::Submission::MonitoringService.for(@submission, @assessment)
+  end
+
+  def should_monitor?
```

Comment:
Did you mean `view_password_protected?`?

Actually, there's no need to. That's the purpose of the `for(submission, assessment)` method in `Course::Assessment::Submission::MonitoringService`. It will only initialise the monitoring service if there is a `Course::Monitoring::Monitor` available for that submission's assessment.

Since `Course::Assessment::AssessmentsController` only creates a monitor if the assessment is password-protected (i.e., an exam), there shouldn't be a way for a non-exam submission to start monitoring.

The only exception is if:
1. An assessment A was set to enable exam monitoring,
2. there was a submission S created,
3. before the session expires, assessment A was set to disable exam monitoring, and
4. the submission S page is refreshed.

Then in this case, the submission monitoring service will be initialised, but `listening?` will return `false` because `@monitor.enabled?` is `false`. The subscription to `HeartbeatChannel` will fail and the worker dies.